tests/unit/multitoken_library_test.cairo

```diff
@@ -0,0 +1,290 @@
+        assert_eq!(0, 1);
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_should_fail_when_erc20_when_source_is_this_when_call_to_non_contract_address() {
+        let non_contract_address = BOB();
+        let this_address = starknet::get_contract_address();
+
+        let asset = MultiToken::ERC20(non_contract_address, 1000);
+        asset.transfer_asset_from(this_address, BOB(), false);
+    }
+
+    #[test]
+    fn test_should_call_transfer_when_erc20_when_source_is_not_this() {
+        let token_address = deploy_erc20_mock();
+        let this_address = starknet::get_contract_address();
+        mint_erc20(token_address, ALICE(), 1000);
+
+        assert_eq!(
+            ERC20ABIDispatcher { contract_address: token_address }.balance_of(ALICE()), 1000
+        );
+        assert_eq!(ERC20ABIDispatcher { contract_address: token_address }.balance_of(BOB()), 0);
+
+        give_allowance_erc20(token_address, ALICE(), this_address, 1000);
+
+        let asset = MultiToken::ERC20(token_address, 1000);
+        asset.transfer_asset_from(ALICE(), BOB(), false);
+
+        assert_eq!(ERC20ABIDispatcher { contract_address: token_address }.balance_of(ALICE()), 0);
+        assert_eq!(ERC20ABIDispatcher { contract_address: token_address }.balance_of(BOB()), 1000);
+    }
+
+    #[test]
+    fn test_should_fail_when_erc20_when_source_is_not_this_when_transfer_returns_fale() {
+        // Q - how to make the transfer fail?
+        assert_eq!(0, 1);
+    }
+
+
+    #[test]
+    #[should_panic]
+    fn test_should_fail_when_erc20_when_source_is_not_this_when_call_to_non_contract_address() {
+        let non_contract_address = BOB();
+
+        // Set up a non-contract address as the token address
+        let asset = MultiToken::ERC20(non_contract_address, 1000);
+
+        // Attempt to transfer from ALICE to BOB using the non-contract address
+        // This should panic because the token address is not a valid contract
+        asset.transfer_asset_from(ALICE(), BOB(), false);
+    }
+
+    // ERC721
+
+    #[test]
+    fn test_should_call_safe_transfer_from_when_erc721() {
+        let token_address = deploy_erc721_mock();
+        let this_address = starknet::get_contract_address();
+        mint_erc721(token_address, ALICE(), 1);
+
+        assert_eq!(ERC721ABIDispatcher { contract_address: token_address }.owner_of(1), ALICE());
+
+        let asset = MultiToken::ERC721(token_address, 1);
+
+        give_allowance_erc721(token_address, ALICE(), this_address, 1);
+        asset.transfer_asset_from(ALICE(), BOB(), false);
+
+        assert_eq!(ERC721ABIDispatcher { contract_address: token_address }.owner_of(1), BOB());
+    }
+
+    // ERC1155
+
+    #[test]
+    fn test_should_call_safe_transfer_from_when_erc1155() {
+        // Set up the ERC1155 token
+        let token_address = deploy_erc1155_mock();
+        let this_address = starknet::get_contract_address();
+        mint_erc1155(token_address, ALICE(), 1, 10);
```

Comment:
Getting the following error here. I'm thinking it might be related to the acceptance check performed by `ERC1155.mint_with_acceptance_check()`. According to the [docs](https://docs.openzeppelin.com/contracts-cairo/0.14.0/api/erc1155#ERC1155Component), this should cause an issue only if `starknet::contract_address_const::<'ACCOUNT2'>()` is treated as a contract and not an EOA. Is this the case?

Reply:
![image](https://github.com/NethermindEth/pwn-starknet/assets/16985953/8867c2a0-b1b6-4bf5-bf96-9b79677e9ba9)